### PR TITLE
Switch from gitbook to mdbook

### LINKED ISSUE #256: Migrate away from gitbook
The `gitbook` CLI tool [has been deprecated](https://github.com/GitbookIO/gitbook/commit/6c6ef7f4af32a2977e44dd23d3feb6ebf28970f4). Let's find an alternative, e.g. [`mdbook`](https://rust-lang-nursery.github.io/mdBook/).

### PR DESCRIPTION
**Pull request by flodolo**
Fixes #256

### CODE REVIEW

**Comment by flodolo** [comment]
The current rendering is available here: https://test.flod.org/tmp_guide/

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+jobs:
+  fetch:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v4
+      - name: Install Rust packages
+        run: |
+          curl --proto '=https' --tlsv1.2 -sSf https://rossmacarthur.github.io/install/crate.sh | bash -s -- --repo "rust-lang/mdBook" --bin mdbook --to ~/.cargo/bin
```
Comment (eemeli) [inline .github/workflows/deploy.yml:24]: This seems like a very surprising way to install mdBook.

The user guide includes a section on [Running `mdbook` in Continuous Integration](https://rust-lang.github.io/mdBook/continuous-integration.html), which suggests two different methods, in case we're not happy with just
```sh
cargo install mdbook
```
which should also "just work".

A decent indication that the recommended methods work is that they're used to build the [rust-lang docs site](https://github.com/rust-lang/cargo/blob/7bb7b539558dc88bea44cee4168b6269bf8177b0/.github/workflows/main.yml#L244-L248).

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+          npm install gh-pages@"~6.1.1"
+      - name: Build book
+        run: |
+            mdbook --version
+            mdbook build
+      - name: Set Git config
+        run: |
+          git config --global user.email "actions@users.noreply.github.com"
+          git config --global user.name 'Automation'
+          git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/$GITHUB_REPOSITORY
```
Comment (eemeli) [inline .github/workflows/deploy.yml:36]: Is this last line necessary? If we add a top-level config
```yaml
permissions:
  contents: write
```
would that not have the same effect, without requiring a `GITHUB_TOKEN` getting passed around?

File: .nojekyll
Comment (eemeli) [inline .nojekyll:1]: Why is this added? AFAIK, there are no files or directories starting with an underscore in the mdbook results.

File: package.json [hunk 3]
```diff
@@ -40,8 +40,7 @@
         "eslint": "^6.7.2",
         "@fluent/bundle": "^0.14.0",
         "@fluent/syntax": "^0.14.0",
-        "gh-pages": "^2.1.1",
-        "gitbook-cli": "^2.3.2",
+        "gh-pages": "^6.1.1",
```
Comment (eemeli) [inline package.json:43]: Should probably drop this and the `npm run deploy` script, if/as we're going to presume that the action will take care of that.

**Review by eemeli — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (flodolo) [reply .github/workflows/deploy.yml:24]: This is copied from our existing repositories, it looks like it became obsolete. As a disclaimer, this code came from someone very familiar with Rust and, at the time, the only alternative was compiling from source.

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (flodolo) [reply .github/workflows/deploy.yml:36]: Similarly, not sure this was available at the time. I'll test with the permissions bit.

File: .nojekyll
Comment (flodolo) [reply .nojekyll:1]: We had [other errors](https://github.com/mozilla-l10n/documentation/pull/126#issuecomment-373093676) (again, a long time time ago). Since we don't use Jekyll, I tend to leave it there and disable it.

**Review by flodolo — COMMENTED** [review]
I'll test the changes you suggested on another repo

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (flodolo) [reply .github/workflows/deploy.yml]: It doesn't work, I end up with an error
https://github.com/tschaub/gh-pages/issues/384

I'm guessing that I could create a PAT and use that, but I'd prefer avoiding it since this method works fine.

File: package.json (hunk 3 above)
Comment (flodolo) [reply package.json]: Removed this and the deploy action, but I didn't run any npm command (as said, I'd prefer for someone who understands JS to do that)

**Review by eemeli — APPROVED** [review]
